### DISCUSSION TITLE
Is the Async EntryActionBehaviour broken or am I missing something

### DISCUSSION THREAD

**Discussion by magrym** [dotnet-state-machine/stateless#529] [Q&A]
I moved from the nuget package 5.13.0 to 5.13.0-dev-4941000713 to get the RetainSynchronizationContext feature, but now my async entry functions don't work. I am getting: 
Cannot execute asynchronous action specified in OnEntry event for '***' state. Use asynchronous version of Fire [FireAsync]

And when I look at the code in AsyncFrom.ExecuteAsync in src/Stateless/EntryActionBehaviour.cs it always calls the sync version Execute.

**Comment by mclift** [comment]
Hi @magrym. Thanks for taking the time to report this. I've created #531 to track this issue.
